concent_api/core/message_handlers.py

```diff
@@ -791,26 +803,27 @@ def handle_send_force_payment(
             payment_ts              = payment_ts,
             task_owner_key          = requestor_ethereum_public_key,
             provider_eth_account    = provider_eth_address,
-            amount_paid             = requestors_claim_payment_info.amount_paid,
-            amount_pending          = requestors_claim_payment_info.amount_pending,
+            amount_paid             = claim_against_requestor.amount,  # type: ignore
+            amount_pending          = 0,
             recipient_type          = message.concents.ForcePaymentCommitted.Actor.Provider,
         )
 
         requestor_force_payment_commited = message.concents.ForcePaymentCommitted(
             payment_ts              = payment_ts,
             task_owner_key          = requestor_ethereum_public_key,
             provider_eth_account    = provider_eth_address,
-            amount_paid             = requestors_claim_payment_info.amount_paid,
-            amount_pending          = requestors_claim_payment_info.amount_pending,
+            amount_paid             = claim_against_requestor.amount,  # type: ignore
+            amount_pending          = 0,
             recipient_type          = message.concents.ForcePaymentCommitted.Actor.Requestor,
         )
 
-        store_pending_message(
-            response_type=PendingResponse.ResponseType.ForcePaymentCommitted,
-            client_public_key=requestor_public_key,
-            queue=PendingResponse.Queue.ReceiveOutOfBand,
-            payment_message=requestor_force_payment_commited
-        )
+        with transaction.atomic(using='control'):
+            store_pending_message(
+                response_type=PendingResponse.ResponseType.ForcePaymentCommitted,
+                client_public_key=requestor_public_key,
+                queue=PendingResponse.Queue.ReceiveOutOfBand,
+                payment_message=requestor_force_payment_commited
+            )
 
         provider_force_payment_commited.sig = None
```

Comment:
Is there any reason we are doing this?

Reply:
I can't see any. We sometimes clear (or used to clear) the signature to have it recalculated. If you change something in the message and leave the signature as is, golem-messages won't update it when the message is serialized. But that's not the case here. We're creating a new instance of `ForcePaymentCommitted` so it should not be necessary.